File: easy/Q3/anagram.js
// Function to check the datatype of {value}
const checkType = (value) =>
  typeof value === "string"
    ? value.toLowerCase()
    : value.toString().toLowerCase();

// Function to check if {value} is anagram
const isAnagram = (value) =>
  checkType(value)
    .split("")
    .sort((a, b) => a.localeCompare(b))
    .join("");

console.log(isAnagram("mamba") === isAnagram("AmaMB")); // true
console.log(isAnagram("mamba") === isAnagram("memaB")); // false
console.log(isAnagram(12154) === isAnagram(14521)); // true
console.log(isAnagram(1234) === isAnagram(5678)); // false

File: easy/Q3/anagram.py
def checkType(value):
    """ Function to check {value}'s data-type  """
    return value.lower() if type(value) == str else str(value).lower()


def is_anagram(value):
    """ Function to check {value} is anagram """
    return "".join(sorted(list(checkType(value))))


if __name__ == "__main__":
    print(is_anagram("maMba") == is_anagram("bamam"))  # True
    print(is_anagram("maMba") == is_anagram("memba"))  # False
    print(is_anagram(1234) == is_anagram(1423))  # True
    print(is_anagram(1234) == is_anagram(2315))  # Flase